src/components/feed/AddNode/AddNode.tsx

```diff
@@ -33,12 +35,24 @@ function getInitialState() {
 }
 
 const AddNode: React.FC<AddNodeProps> = ({
+  readOnly,
   selectedPlugin,
   pluginInstances,
   getParams,
   addNode,
   params,
 }: AddNodeProps) => {
+  const currentFeedPayload = useTypedSelector((state) => state.feed.currentFeed)
+  const cookie = new Cookies()
+  const user2 = cookie.get('username')
+  const { data: feed} = currentFeedPayload;
+  if(feed && feed.data.creator_username===user2){
+    readOnly = false
```

Comment:
if `readOnly` is a prop, then it should not be modified. To elaborate:

- if `readOnly` is a prop, then it is the parent's responsibility to evaluate the logic `feed.data.creator_username===user2`
- if `readOnly` is a state, then it is this component's responsibility to evaluate the logic `feed.data.creator_username===user2`